app/code/Magento/Catalog/Model/ResourceModel/Category/Collection.php

```diff
@@ -327,7 +328,7 @@ public function loadProductCount($items, $countRegular = true, $countAnchor = tr
             $countSelect = $this->getProductsCountQuery($categoryIds, (bool)$websiteId);
             $categoryProductsCount = $this->_conn->fetchPairs($countSelect);
             foreach ($anchor as $item) {
-                $productsCount = isset($categoriesProductsCount[$item->getId()])
+                $productsCount = isset($categoryProductsCount[$item->getId()])
```

Comment:
This change which was mentioned in initial ticket sure makes sense.
Could you please improve it even more with

```
$productsCount = (int)($categoryProductsCount[$item->getId()] ?? $this->getProductsCountFromCategoryTable($item, $websiteId));
```

Reply:
That's what I tried first. But as I was trying to explain, `getProductsCountQuery` (which is supposed to fill `$categoryProductsCount`) try to get data from `catalog_category_product_index`, which is an obsolete table which stays empty... That table has been replaced by `catalog_category_product_index_storeX`, leaving no more general index table for admin...

If you keep only the change in Collection class, so the variable rename, and using `$tableMaintainer` in `getProductsCountQuery` method, then we'll have a partial fix that solves the issue when user selects a store view in backend category page. But the issue will remains on default view. That's why I added mechanics for `catalog_category_product_index_store0`...

Reply:
@slopukhov I see you're the one who made that change (a long time ago ^^).. Maybe you have an opinion on this and a better solution that the one I proposed ?